oci/layout/oci_dest.go

```diff
@@ -133,8 +133,12 @@ func (d *ociImageDestination) PutBlob(stream io.Reader, inputInfo types.BlobInfo
 	if err := blobFile.Sync(); err != nil {
 		return types.BlobInfo{}, err
 	}
-	if err := blobFile.Chmod(0644); err != nil {
-		return types.BlobInfo{}, err
+
+	// Chmod is not supported on Windows
+	if runtime.GOOS != "windows" {
+		if err := blobFile.Chmod(0644); err != nil {
```

Comment:
How does `Chmod` fail here?  Reading `go/src/os/file_posix.go` and `go/src/syscall/syscall_windows.go` it seems to be translated to `GetFileAttributes`+`SetFileAttributes` modifying the readonly attribute, which seems plausible.

Reply:
There is a difference between os.Chmod and os.File.Chmod. Go figure. See also here https://github.com/google/skicka/issues/77. As a workaround, we can also try to use os.Chmod at this point.

Reply:
Oops, missed that.  The code is fine, please just add a more detailed explanation, something like
```go
// On POSIX systems, blobFile was created with mode 0600, so we need to make it readable.
// On Windows, the “permissions of newly created files” argument to syscall.Open is
// ignored and the file is already readable; besides, blobFile.Chmod, i.e. syscall.Fchmod,
// always fails on Windows.
```